jpa/org.eclipse.persistence.jpa/src/org/eclipse/persistence/internal/jpa/metadata/accessors/objects/MetadataAnnotatedElement.java

```diff
@@ -202,17 +204,36 @@ public MetadataAnnotation getAnnotation(Class annotation) {
     /**
      * INTERNAL:
      * Return the annotated element for this accessor. Note: This method does
-     * not check against a metadata complete.
+     * not check against a meta-data complete.
+     * @param annotation annotation name
+     */
+    public MetadataAnnotation getAnnotation(final String annotation) {
+        return getAnnotation(annotation, new HashSet<>());
+    }
+
+    /**
+     * INTERNAL:
+     * Return the annotated element for this accessor. Note: This method does
+     * not check against a meta-data complete.
+     * @param annotation annotation name
+     * @param names meta-annotations names cycle detection set, shall not be {@code null}
      */
-    public MetadataAnnotation getAnnotation(String annotation) {
+    protected final MetadataAnnotation getAnnotation(final String annotation, final Set<String> names) {
         if (m_annotations == null && m_metaAnnotations == null) {
             return null;
         }
-
         MetadataAnnotation metadataAnnotation = m_annotations.get(annotation);
         if (metadataAnnotation == null) {
             for (MetadataAnnotation a: m_metaAnnotations.values()) {
-                MetadataAnnotation ma = m_factory.getMetadataClass(a.getName()).getAnnotation(annotation);
+                if (names.contains(a.getName())) {
```

Comment:
@cristhiank I changed the code to always break the loop when there is a cycle. Unfortunately javax.persistence annotation won't work. Can you add some jUnit based on your own meta-annotations to verify this scenario?

Reply:
@Tomas-Kraus I can confirm that it is working now with non-existent annotations. I hope this fix gets merged soon. Thanks!